### PR TITLE
Added docs for raw_ops functions aliasing

### PR DESCRIPTION
**Pull request by hirwa-nshuti**
Also Added the missing `@to_ivy_arrays_and_back` decorator on the example functions in the docs
Thanks:)

### CODE REVIEW

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 1]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
     def tan(x, name=None):
         return ivy.tan(x)
 
 Likewise, :code:`tan` is also placed under :mod:`math`.
 By referring to the `tf.math.tan`_ documentation, we add the same arguments, and simply wrap :func:`ivy.tan` in this case.
 Again, we do not support the :code:`name` argument for the reasons outlined above.
 
+**NOTE**
+
+For The :mod:`tf.raw_ops` many functions posses the same behavior as those of :mod:`tf` namespace except that only the functions from :mod:`tf.raw_ops` takes key-word only arguments.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:218]: Those in the `tf` namespace

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 2]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+For this we have a designed decorator to wrap the :mod:`tf` namespace function to satisfy this behavior. Let's take an example of :func:`tf.math.argmax` and :func:`tf.raw_ops.ArgMax`.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:219]: To tackle this disparity in behaviour, the decorator `@map_raw_ops_alias` was designed to wrap these functions that exist in the TensorFlow namespace and reduce the number of implementations necessary.

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 3]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+    
+    # in ivy/functional/frontends/tensorflow/math.py
+    @to_ivy_arrays_and_back
+    def argmax(input, axis, output_type=None, name=None):
+        if output_type in ["uint16", "int16", "int32", "int64"]:
+            return ivy.astype(ivy.argmax(input, axis=axis), output_type)
+        else:
+            return ivy.astype(ivy.argmax(input, axis=axis), "int64")
+
+This function is a frontend ivy implementation of :func:`tf.math.argmax`, the similar function in :mod:`tf.raw_ops` has similar inputs and outputs except the :func:`tf.raw_ops.ArgMax` takes all inputs as key-word only.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:231]: This function,` argmax()` is implemented in `tf.math.argmax`. There exists another `argmax()` within the TensorFlow framework, specifically in the `tf.raw_ops submodule. This function has identical behaviour with the exception that all arguments for this function are keyword only.

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 4]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+Let's take a look on how we can handle such case.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:232]: Let's see how we can use `map_raw_ops_alias` to handle this difference

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 5]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+
+.. code-block:: python
+
+    # in ivy/functional/frontends/tensorflow/raw_ops.py
+    ArgMax = to_ivy_arrays_and_back(
+        map_raw_ops_alias(
+            tf_frontend.math.argmax,
+            kwargs_to_update={"dimension": "axis"},
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:240]: I would explain that this is a dictionary, and how to use it in the general sense

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 6]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+        )
+    )
+
+The :func:`tf_frontend.func_wrapper.map_raw_ops_alias` is the one that is wrapping the behavior of :func:`tf_frontend.math.argmax`, change it behavior to receive key-word only arguments.
+In case you are curious to know more about this function wrapper you can find it here: `map_raw_ops_alias <https://github.com/unifyai/ivy/blob/54cc9cd955b84c50a1743dddddaf6e961f688dd5/ivy/functional/frontends/tensorflow/func_wrapper.py#L127>`_
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:245]: The wrapper is implemented here <> in the Ivy codebase.

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Just some edit suggestions to improve readability. Some of the explanations were hard to follow at first glance, but maybe that's just me :). Also, I would explain how to use the wrapper in the general sense, possibly through a breakdown or arguments.

**Comment by hirwa-nshuti** [comment]
> > Just some edit suggestions to improve readability. Some of the explanations were hard to follow at first glance, but maybe that's just me :). Also, I would explain how to use the wrapper in the general sense, possibly through a breakdown or arguments.
> 
> Also, apologies if the review felt a bit harsh. It felt easier to quickly rephrase stuff rather than explain. Feel free to take as inspiration and make improvements on your own accord :)

Yea it's a sure thing and I strongly agree with you let me work it out.
Thanks 🙂

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 7]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
     def tan(x, name=None):
         return ivy.tan(x)
 
 Likewise, :code:`tan` is also placed under :mod:`math`.
 By referring to the `tf.math.tan`_ documentation, we add the same arguments, and simply wrap :func:`ivy.tan` in this case.
 Again, we do not support the :code:`name` argument for the reasons outlined above.
 
+**NOTE**
+
+Many of the functions in the :mod:`tf.raw_ops` module have identical behaviuor to the functions that are generally in the TensorFlow namespace e.g :func:`tf.argmax`, with exception that the functions in the function in the :mod:`tf.raw_ops` module are specified to have key-word only arguments and in some case there is a naming difference of arguments.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:218]: behaviuor - behaviour

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:218]: with exception that the functions in the function in the - with the exception that functions in the module, :mod:`tf.raw_ops`, are

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:218]: in some case there is a naming difference of arguments - in some cases there is a difference in argument naming

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 8]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+    
+    # in ivy/functional/frontends/tensorflow/math.py
+    @to_ivy_arrays_and_back
+    def argmax(input, axis, output_type=None, name=None):
+        if output_type in ["uint16", "int16", "int32", "int64"]:
+            return ivy.astype(ivy.argmax(input, axis=axis), output_type)
+        else:
+            return ivy.astype(ivy.argmax(input, axis=axis), "int64")
+
+This function :func:`argmax` is implemented in the :mod:`tf.math` module of the TensoFlow framework, there exists identical function in the :mod:`tf.raw_ops` module implemented as :func:`ArgMax`. Both the functions have identical behaviour except that all arguments are passed as key-word only for :func:`tf.raw_ops.ArgMax` and in some corner cases, arguments are renamed where :code:`dimension` argument replaces :code:`axis` in case of the :func:`tf.math.argmax` function.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:231]: there exists identical - > there exists an identical

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:231]: Both the functions have identical behaviour except that all -> both the functions have identical behaviour except for the fact that all

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:231]: arguments are renamed where :code:`dimension` argument replaces :code:`axis` in case of the :func:`tf.math.argmax` function. -> such as the case in :func:`tf.math.argmax`, the :code:`dimension` argument replaces :code:`axis`.

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 9]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+
+    # in ivy/functional/frontends/tensorflow/raw_ops.py
+    ArgMax = to_ivy_arrays_and_back(
+        map_raw_ops_alias(
+            tf_frontend.math.argmax,
+            kwargs_to_update={"dimension": "axis"},
+        )
+    )
+
+The decorator :code:`map_raw_ops_alias` here takes the existing behaviour of :func:`tf_frontend.math.argmax` as it's first parameter, and changes all it's arguments to key-word only, with consideration of the argument :code:`kwargs_to_update` passed to it as a dictionary indicating all updates in arguments names to be made, as for the case of :code:`dimension` replacing to :code:`axis` in the :func:`tf.raw_ops.ArgMax` function.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:244]: decorator :code:`map_raw_ops_alias` here(,) takes

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:244]: f :func:`tf_frontend.math.argmax` as **it's** first parameter - f :func:`tf_frontend.math.argmax` as **its** first parameter

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:244]: and changes all it's arguments to key-word only, -> and changes all it's arguments to key-word only.

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:244]: with consideration of the argument :code:`kwargs_to_update` passed to it as a dictionary indicating all updates in arguments names to be made -> :code:`kwargs_to_update` is a dictionary indicating all updates in arguments names to be made.

Comment (jkeane508) [reply docs/partial_source/deep_dive/ivy_frontends.rst:244]: as for the case of :code:`dimension` replacing to :code:`axis` in the :func:`tf.raw_ops.ArgMax` function. -> in the case of :func:`tf.raw_ops.ArgMax` , :code:`dimension` is replaced with :code:`axis`.

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Just some changes needed to improve readability :)

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 10]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+    
+    # in ivy/functional/frontends/tensorflow/math.py
+    @to_ivy_arrays_and_back
+    def argmax(input, axis, output_type=None, name=None):
+        if output_type in ["uint16", "int16", "int32", "int64"]:
+            return ivy.astype(ivy.argmax(input, axis=axis), output_type)
+        else:
+            return ivy.astype(ivy.argmax(input, axis=axis), "int64")
+
+This function :func:`argmax` is implemented in the :mod:`tf.math` module of the TensorFlow framework, there exists an identical function in the :mod:`tf.raw_ops` module implemented as :func:`ArgMax`. Both the functions have identical behaviour except the fact that all arguments are passed as key-word only for :func:`tf.raw_ops.ArgMax` and in some corner cases, arguments are renamed such in the case of :func:`tf.math.argmax`, the :code:`dimension` argument replaces :code:`axis`.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:231]:
```suggestion
This function :func:`argmax` is implemented in the :mod:`tf.math` module of the TensorFlow framework, there exists an identical function in the :mod:`tf.raw_ops` module implemented as :func:`ArgMax`. Both the functions have identical behaviour except for the fact that all arguments are passed as key-word only for :func:`tf.raw_ops.ArgMax`. In some corner cases, arguments are renamed such as :func:`tf.math.argmax`, the :code:`dimension` argument replaces the :code:`axis` argument. 
```

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 11]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
     def tan(x, name=None):
         return ivy.tan(x)
 
 Likewise, :code:`tan` is also placed under :mod:`math`.
 By referring to the `tf.math.tan`_ documentation, we add the same arguments, and simply wrap :func:`ivy.tan` in this case.
 Again, we do not support the :code:`name` argument for the reasons outlined above.
 
+**NOTE**
+
+Many of the functions in the :mod:`tf.raw_ops` module have identical behaviour to the functions that are generally in the TensorFlow namespace e.g :func:`tf.argmax`, with exception that the functions in the module :mod:`tf.raw_ops` are specified to have key-word only arguments and in some cases there is a difference of argument naming.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:218]:
```suggestion
Many of the functions in the :mod:`tf.raw_ops` module have identical behaviour to functions in the general TensorFlow namespace e.g :func:`tf.argmax`, with the exception of functions in the module :mod:`tf.raw_ops`. However, these functions are specified to have key-word only arguments and in some cases they have different argument names.
```

File: docs/partial_source/deep_dive/ivy_frontends.rst [hunk 12]
```diff
@@ -199,18 +205,51 @@ Ivy does not support the unique naming of individual operations, and so we omit
...
+
+    # in ivy/functional/frontends/tensorflow/raw_ops.py
+    ArgMax = to_ivy_arrays_and_back(
+        map_raw_ops_alias(
+            tf_frontend.math.argmax,
+            kwargs_to_update={"dimension": "axis"},
+        )
+    )
+
+The decorator :code:`map_raw_ops_alias` here, takes the existing behaviour of :func:`tf_frontend.math.argmax` as its first parameter, and changes all its arguments to key-word only, with consideration of the argument :code:`kwargs_to_update` is a dictionary indicating all updates in arguments names to be made, in the case of :func:`tf.raw_ops.ArgMax`, :code:`dimension` is replacing :code:`axis`.
```
Comment (jkeane508) [inline docs/partial_source/deep_dive/ivy_frontends.rst:244]:
```suggestion
The decorator :code:`map_raw_ops_alias` here, takes the existing behaviour of :func:`tf_frontend.math.argmax` as its first parameter, and changes all its arguments to key-word only. The argument :code:`kwargs_to_update` is a dictionary indicating all updates in arguments names to be made, in the case of :func:`tf.raw_ops.ArgMax`, :code:`dimension` is replacing :code:`axis`.
```

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Almost there by the looks of it :)

**Review by jkeane508 — APPROVED** [review]
LGTM!
